### ISSUE TITLE
Type 'null' / `undefined` is not assignable to type 'IQueryResult' when `strict: true`

### ISSUE THREAD

**Issue by kaorun343** [nestjs/cqrs#267]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

```ts
@Injectable()
export class MyService {
  constructor(private queryBus: QueryBus) {}

  async getBook(bookId: string): Promise<Book | null> {
    return this.queryBus.execute<GetBookQuery, Book | null>(new GetBookQuery(bookId))
  }
}
```

```
error TS2344: Type 'Book | null' does not satisfy the constraint 'IQueryResult'.
Type 'null' is not assignable to type 'IQueryResult'
```

This error happens not only with `null` but also `undefined`.

## Expected behavior

Even if TypeScript compiler is in strict mode, `null` and `undefined` should be able to set as the return value of `QueryBus#execute`.

## Environment

<pre><code>
Nest core version: 7.0.0
Nest CQRS version: 6.1.0

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by underfisk** [comment]
Hi, did you try removing <GetBookQuery, Book | null> because Book or null is already specified in your Promise so it's kinda redundant?
I think that is also the issue, you simply need to execute the query, there's no need to specify the type, nest will gather it automatically

**Comment by kamilmysliwiec** [comment]
This should be fixed in v7.0.0
